locker/memory_locker.go

```diff
@@ -0,0 +1,125 @@
+package locker
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+type MemoryKey struct {
+	id             string
+	key            string
+	expirationTime time.Time
+	Channel        *chan error
+}
+
+func (k *MemoryKey) ID() string {
+	return k.id
+}
+
+func (k *MemoryKey) Key() string {
+	return k.key
+}
+
+func (k *MemoryKey) ExpirationTime() time.Time {
+	return k.expirationTime
+}
+
+func (k *MemoryKey) SetExpirationTime(t time.Time) error {
+	k.expirationTime = t
+	return nil
+}
+
+type MemoryLocker struct {
+	lockedItems sync.Map
+	mutex       *sync.Mutex
+}
+
+func (m *MemoryLocker) Lock(key string) (Key, error) {
+	if key == "" {
+		return &MemoryKey{}, fmt.Errorf("key is empty")
+	}
+
+	id := uuid.New().String()
+
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+
+	if lockInfo, ok := m.lockedItems.Load(key); ok {
+		keyLock := lockInfo.(LockInformation)
+		if time.Since(keyLock.Date) < ValidTimePeriod {
+			return &MemoryKey{}, fmt.Errorf("key '%s' is already locked by: %s", key, keyLock.ID)
+		}
+	}
+
+	lockChannel := make(chan error)
+	lockKey := MemoryKey{id: id, key: key, Channel: &lockChannel}
+
+	lock := LockInformation{
+		ID:   id,
+		Key:  key,
+		Date: time.Now().UTC(),
+	}
+	m.lockedItems.Store(key, lock)
+
+	go m.updateLockTime(id, key, lockChannel)
+
+	return &lockKey, nil
+}
+
+func (m *MemoryLocker) updateLockTime(id string, key string, lockChannel chan error) {
```

Comment:
nit: use the "arrow" syntax to indicate this only reads from the channel

i.e. `lockChannel <-chan error`